src/syslog/syslog.c

```diff
@@ -64,3 +63,28 @@ void syslog_transport_close(syslog_transport_t *transport) {
     }
     transport->is_open = false;
 }
+
+bool syslog_transport_open(syslog_transport_t *transport, const char *host, int port) {
+    ip_addr_t addr;
+    err_t res = dns_gethostbyname(host, &addr, NULL, NULL);
+    if (res == ERR_OK) {
+        // ip address already cached or the name was valid ip address
+        if (transport->last_resolve_state != Resolved) {
+            char ip_addres[16];
+            ip4addr_ntoa_r(&addr, ip_addres, 16);
+            log_info(Syslog, "Host name %s resolved to this ip: %s", host, ip_addres);
+            transport->last_resolve_state = Resolved;
+        }
+        return syslog_transport_open_ip4(transport, addr, port);
+    }
+    if (res == ERR_INPROGRESS) {
+        // dns lookup queued, wait for it to finish
+        log_info(Syslog, "DNS resolving for host %s was enqueued", host);
+        transport->last_resolve_state = Progress;
+        return false;
+    }
```

Comment:
I wonder if this will always converge to success. I assume you try to look again from time to time and rely on having a working DNS cache. Nevertheless:

* Do we _have_ the cache? How large?
* What if other tasks also resolve and kick this one out?
* What if the DNS record has TTL 0 and forbids caching?

Reply:
Yes we have the cache iand it stores two IPs. 
If the IP gets kicked out, it will be queried once more and kickout another IP
If it has forbiden caching we wont get the IP.

Reply:
I guess I won't block it over these concerns… this is debugging aid, not really production code. But the fact we abuse the cache this way (eg. try again later instead of using the callback) seems a bit wrong.